src/detect-engine-alert.c

```diff
@@ -179,7 +179,7 @@ static void PacketApplySignatureActions(Packet *p, const Signature *s, const uin
     SCLogDebug("packet %" PRIu64 " sid %u action %02x alert_flags %02x", p->pcap_cnt, s->id,
             s->action, alert_flags);
 
-    if (s->action & ACTION_DROP) {
+    if (s->action & (ACTION_DROP | ACTION_REJECT_ANY)) {
```

Comment:
will this not just treat rejects as drop?

Reply:
My local fix is to add `PACKET_UPDATE_ACTION(p, s->action);` below the `PacketDrop`

Reply:
> will this not just treat rejects as drop?

Oh, sorry, I misunderstand how the `reject` action works >__<'

Reply:
> My local fix is to add `PACKET_UPDATE_ACTION(p, s->action);` below the `PacketDrop`

thanks! :)

Reply:
> My local fix is to add `PACKET_UPDATE_ACTION(p, s->action);` below the `PacketDrop`

Do you mean like this?
```diff
- if (s->action & (ACTION_DROP) {
-        PacketDrop(p, PKT_DROP_REASON_RULES);
-
+ if (s->action & (ACTION_DROP | ACTION_REJECT_ANY)) {
+        PacketDrop(p, PKT_DROP_REASON_RULES);
+        PACKET_UPDATE_ACTION(p, s->action);
```
If that is the case, I wonder if we could call `PACKET_UPDATE_ACTION` before the checks, since it would be called in both arms...

Reply:
> > My local fix is to add `PACKET_UPDATE_ACTION(p, s->action);` below the `PacketDrop`
> 
> Do you mean like this?
> 
> ```diff
> - if (s->action & (ACTION_DROP) {
> -        PacketDrop(p, PKT_DROP_REASON_RULES);
> -
> + if (s->action & (ACTION_DROP | ACTION_REJECT_ANY)) {
> +        PacketDrop(p, PKT_DROP_REASON_RULES);
> +        PACKET_UPDATE_ACTION(p, s->action);
> ```
> 
> If that is the case, I wonder if we could call `PACKET_UPDATE_ACTION` before the checks, since it would be called in both arms...

No, if you take a look at the PacketDrop implementation, it calls PACKET_SET_ACTION, which resets the action to DROP.
Ideally, I guess PacketDrop should really call PACKET_UPDATE_ACTION, and then you can move the PACKET_UPDATE_ACTION back to the top of the function.